### PR TITLE
[HW11] Add `DestroyProject` mutation

### PR DESCRIPTION
**Pull request by w1nway**
### Summary

A brief description of the pull request.

### How it works

Screenshots/screencasts of the pull request introduced functionality.

### Test plan

List of steps to manually test introduced functionality:

- Go to http://localhost:3000/graphiql
- Make request using schema:
```graphql
  query {
    me {
      id
      name
    }
  }
```

### Deploy notes

Notes regarding deployment the contained body of work.
These should note any db migrations, services, scripts, etc.

### Checklist:

- [ ] I have performed a self-review of my own code
- [ ] My pull request is shorter than 500 lines

### CODE REVIEW

**Review by Raily01 — COMMENTED** [review]
## **pretty good, man**

File: app/graphql/mutations/destroy_project.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  class DestroyProject < BaseMutation
+    include GraphqlErrors
+
+    argument :input, Types::Inputs::DestroyProjectInput, required: true
+
+    type Types::Payloads::DestroyProjectPayload
+
+    def resolve(input:)
+      project = Project.find(input.id)
+      result = Projects::Destroy.call(project: project, project_params: input.to_h, user: current_user)
```
Comment (RasimKhusaenov) [inline app/graphql/mutations/destroy_project.rb:11]: Do you have project params here?

File: app/graphql/types/inputs/destroy_project_input.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+module Types
+  module Inputs
+    class DestroyProjectInput < Types::BaseInputObject
+      argument :id, ID, required: true
```
[the remark this answers is not in the document]
Comment (RasimKhusaenov) [reply app/graphql/types/inputs/destroy_project_input.rb:4]: @Raily01, I don't get what your point is

File: app/graphql/types/payloads/create_project_payload.rb [hunk 3]
```diff
@@ -2,7 +2,6 @@ module Types
   module Payloads
     class CreateProjectPayload < Types::BaseObject
       field :project, ProjectType, null: true
-      field :errors, [Types::UserError], null: false
```
[the remark this answers is not in the document]
Comment (w1nway) [reply app/graphql/types/payloads/create_project_payload.rb:5]: Errors mustn't show after destroy actioin

File: app/graphql/types/inputs/destroy_project_input.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (w1nway) [reply app/graphql/types/inputs/destroy_project_input.rb:4]: it' s simple: @Raily01 want 'dokopats'a'

File: app/graphql/mutations/destroy_project.rb (hunk 1 above)
Comment (w1nway) [reply app/graphql/mutations/destroy_project.rb:11]: I think that isn't necessary

File: app/graphql/types/payloads/create_project_payload.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (RasimKhusaenov) [reply app/graphql/types/payloads/create_project_payload.rb:5]: Okay, this payload is used for what?

File: app/graphql/mutations/destroy_project.rb (hunk 1 above)
Comment (RasimKhusaenov) [reply app/graphql/mutations/destroy_project.rb:11]: Then why you you put project params to interactor context?

Comment (w1nway) [reply app/graphql/mutations/destroy_project.rb]: it was redundant

File: app/graphql/types/payloads/destroy_project_payload.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+module Types
+  module Payloads
+    class DestroyProjectPayload < Types::BaseObject
+      field :project, ProjectType, null: true
```
Comment (RasimKhusaenov) [inline app/graphql/types/payloads/destroy_project_payload.rb:4]: Can it be null?

**Review by RasimKhusaenov — CHANGES_REQUESTED** [review]

File: app/graphql/mutations/destroy_project.rb [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    type Types::Payloads::DestroyProjectPayload
+
+    def resolve(input:)
+      project = Project.find(input.id)
+      result = Projects::Destroy.call(project: project, user: current_user)
+
+      if result.success?
+        result.to_h
+      else
+        result.to_h.merge(errors: formatted_errors(result.project))
```
[the remark this answers is not in the document]
Comment (w1nway) [reply app/graphql/mutations/destroy_project.rb:16]: it's my mistake

File: app/graphql/mutations/destroy_project.rb [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+  class DestroyProject < BaseMutation
+    include AuthenticableGraphqlUser
+
+    argument :input, Types::Inputs::DestroyProjectInput, required: true
+
+    type Types::Payloads::DestroyProjectPayload
+
+    def resolve(input:)
+      project = Project.find(input.id)
+      result = Projects::Destroy.call(project: project, user: current_user)
```
[the remark this answers is not in the document]
Comment (w1nway) [reply app/graphql/mutations/destroy_project.rb:11]: yes, i think that user.

File: app/graphql/types/payloads/destroy_project_payload.rb (hunk 4 above)
Comment (w1nway) [reply app/graphql/types/payloads/destroy_project_payload.rb:4]: no))

File: app/graphql/mutations/destroy_project.rb (hunk 6 above)
[the remark this answers is not in the document]
Comment (RasimKhusaenov) [reply app/graphql/mutations/destroy_project.rb:11]: ?

File: app/graphql/mutations/destroy_project.rb (hunk 5 above)
[the remark this answers is not in the document]
Comment (RasimKhusaenov) [reply app/graphql/mutations/destroy_project.rb:16]: @w1nway

File: app/graphql/mutations/destroy_project.rb (hunk 6 above)
[the remark this answers is not in the document]
Comment (w1nway) [reply app/graphql/mutations/destroy_project.rb]: user instead current_user

[the remark this answers is not in the document]
Comment (RasimKhusaenov) [reply app/graphql/mutations/destroy_project.rb]: Why you need user in this interactor 🫠

File: app/graphql/mutations/destroy_project.rb (hunk 5 above)
[the remark this answers is not in the document]
Comment (RasimKhusaenov) [reply app/graphql/mutations/destroy_project.rb:16]: Project doesn't have errors here, it it redundant

File: app/graphql/mutations/destroy_project.rb (hunk 6 above)
[the remark this answers is not in the document]
Comment (w1nway) [reply app/graphql/mutations/destroy_project.rb]: it's not necessary, yes

File: app/graphql/mutations/destroy_project.rb [hunk 7]
```diff
@@ -9,12 +9,7 @@ class DestroyProject < BaseMutation
     def resolve(input:)
       project = Project.find(input.id)
       result = Projects::Destroy.call(project: project)
-
-      if result.success?
-        result.to_h
-      else
-        result.to_h.merge(errors: formatted_errors(result.project))
-      end
+      result.to_h
```
Comment (RasimKhusaenov) [inline app/graphql/mutations/destroy_project.rb:12]:
```suggestion
      Projects::Destroy.call(project: project).to_h
```

**Review by RasimKhusaenov — APPROVED** [review]

**Pull request merged**